apple/Sources/FerrostarSwiftUI/Views/InstructionsView.swift

```diff
@@ -24,72 +30,111 @@ public struct InstructionsView: View {
     ///   - visualInstruction: The visual instruction to display.
     ///   - distanceFormatter: The formatter which controls distance localization.
     ///   - distanceToNextManeuver: The distance remaining for the step.
+    ///   - remainingSteps: All steps remaining in the route, including the current step.
     ///   - primaryRowTheme: The theme for the primary instruction.
     ///   - secondaryRowTheme: The theme for the secondary instruction.
-    ///   - showPillControl: If true, shows a pill control (to indicate an action/expansion).
+    ///   - isExpanded: Whether the instruction view is currently expanded.
+    ///   - sizeWhenNotExpanded: The size of the InstructionsView when minimized. You may use this for allocating space
+    /// for the instruction view in your layout. This property is automatically updated by the instruction view as its
+    /// size changes.
     public init(
         visualInstruction: VisualInstruction,
         distanceFormatter: Formatter = DefaultFormatters.distanceFormatter,
         distanceToNextManeuver: CLLocationDistance? = nil,
+        remainingSteps: [RouteStep]? = nil,
         primaryRowTheme: InstructionRowTheme = DefaultInstructionRowTheme(),
         secondaryRowTheme: InstructionRowTheme = DefaultSecondaryInstructionRowTheme(),
-        showPillControl: Bool = false
+        isExpanded: Binding<Bool> = .constant(false),
+        sizeWhenNotExpanded: Binding<CGSize> = .constant(.zero)
     ) {
         self.visualInstruction = visualInstruction
         self.distanceFormatter = distanceFormatter
         self.distanceToNextManeuver = distanceToNextManeuver
+        self.remainingSteps = remainingSteps
         self.primaryRowTheme = primaryRowTheme
         self.secondaryRowTheme = secondaryRowTheme
-        self.showPillControl = showPillControl
+        _isExpanded = isExpanded
+        _sizeWhenNotExpanded = sizeWhenNotExpanded
     }
 
-    public var body: some View {
-        VStack {
+    private var nextVisualInstructions: [(VisualInstruction, RouteStep)] {
+        guard let remainingSteps, !remainingSteps.isEmpty else {
```

Comment:
Just noticed a subtle bug here. We try to index from 1 below, so we need to look at the actual count.